[scheduler] Crash when creating state

**Describe the bug**
Got this crash:
```
{"level":"info","ts":"2021-09-20T23:53:28.151Z","logger":"kafka-controller","caller":"statefulset/scheduler.go:166","msg":"scheduling","key":"sources-kafka-sanity/fruits"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1d8 pc=0x1a6380d]

goroutine 288 [running]:
knative.dev/eventing-kafka/pkg/common/scheduler/state.(*stateBuilder).State(0xc0002373b0, 0xc0001b9800, 0x0, 0x0, 0xc003adca30)
        knative.dev/eventing-kafka/pkg/common/scheduler/state/state.go:285 +0xeed
knative.dev/eventing-kafka/pkg/common/scheduler/statefulset.(*StatefulSetScheduler).scheduleVPod(0xc0000bcf00, 0x22eb5e8, 0xc001697680, 0x1, 0xc00291b220, 0x2, 0x2, 0x22eb5e8)
        knative.dev/eventing-kafka/pkg/common/scheduler/statefulset/scheduler.go:170 +0x1b8
knative.dev/eventing-kafka/pkg/common/scheduler/statefulset.(*StatefulSetScheduler).Schedule(0xc0000bcf00, 0x22eb5e8, 0xc001697680, 0x0, 0x0, 0x0, 0x0, 0x0)
        knative.dev/eventing-kafka/pkg/common/scheduler/statefulset/scheduler.go:149 +0x291
knative.dev/eventing-kafka/pkg/source/reconciler/mtsource.(*Reconciler).reconcileMTReceiveAdapter(0xc0002fc3f0, 0xc001697680, 0xc000194dc0, 0x4)
        knative.dev/eventing-kafka/pkg/source/reconciler/mtsource/kafkasource.go:164 +0x65
knative.dev/eventing-kafka/pkg/source/reconciler/mtsource.(*Reconciler).ReconcileKind(0xc0002fc3f0, 0x22eacb8, 0xc002963350, 0xc001697680, 0x0, 0x0)
        knative.dev/eventing-kafka/pkg/source/reconciler/mtsource/kafkasource.go:150 +0x17fc
knative.dev/eventing-kafka/pkg/client/injection/reconciler/sources/v1beta1/kafkasource.(*reconcilerImpl).Reconcile(0xc0000bcd20, 0x22eacb8, 0xc002963350, 0xc0016af9e0, 0x1b, 0xc0038f1ed0, 0x22eacb8)
        knative.dev/eventing-kafka/pkg/client/injection/reconciler/sources/v1beta1/kafkasource/reconciler.go:239 +0x13b7
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc0001818c0, 0x0)
        knative.dev/pkg@v0.0.0-20210914164111-4857ab6939e3/controller/controller.go:542 +0x5ef
knative.dev/pkg/controller.(*Impl).RunContext.func3(0xc00284fdc0, 0xc0001818c0)
        knative.dev/pkg@v0.0.0-20210914164111-4857ab6939e3/controller/controller.go:478 +0x53
created by knative.dev/pkg/controller.(*Impl).RunContext
        knative.dev/pkg@v0.0.0-20210914164111-4857ab6939e3/controller/controller.go:476 +0x1a5
```

This is happening because there are placements and the number of kafkasource-mt-adapter replicas is zero. It should not be set zero when reapplying the  kafkasource-mt-adapter yaml file. 

 **Expected behavior**
No crash

 **To Reproduce**
Create a KafkaSource, wait for it to be ready. Re-install MTKafkaSource which sets the number of statefulset replicas to 0

 **Knative release version**

 **Additional context**
Add any other context about the problem here such as proposed priority

--- a/config/source/multi/deployments/adapter.yaml
+++ b/config/source/multi/deployments/adapter.yaml
@@ -20,7 +20,7 @@ metadata:
   labels:
     kafka.eventing.knative.dev/release: devel
 spec:
-  replicas: 0
+  #replicas field removed so that a kafka upgrade doesn't modify the number of replicas after placements have been made
   selector:
     matchLabels: &labels
       control-plane: kafkasource-mt-adapter

--- a/pkg/common/scheduler/state/state.go
+++ b/pkg/common/scheduler/state/state.go
@@ -235,24 +235,26 @@ func (s *stateBuilder) State(reserved map[types.NamespacedName]map[string]int32)
 			return err == nil, nil
 		})
 
-		var unschedulable bool
-		annotVal, ok := pod.ObjectMeta.Annotations[constants.PodAnnotationKey] //Pod is marked for eviction - CANNOT SCHEDULE VREPS on this pod
-		if ok {
-			unschedulable, _ = strconv.ParseBool(annotVal)
-		}
+		if pod != nil {
+			var unschedulable bool
+			annotVal, ok := pod.ObjectMeta.Annotations[constants.PodAnnotationKey] //Pod is marked for eviction - CANNOT SCHEDULE VREPS on this pod
+			if ok {
+				unschedulable, _ = strconv.ParseBool(annotVal)
+			}
 
-		nodeName := pod.Spec.NodeName
-		node, err := s.nodeLister.Get(nodeName) //Node could be marked as Unschedulable - CANNOT SCHEDULE VREPS on a pod running on this node
-		if err != nil {
-			return nil, err
-		}
+			nodeName := pod.Spec.NodeName
+			node, err := s.nodeLister.Get(nodeName) //Node could be marked as Unschedulable - CANNOT SCHEDULE VREPS on a pod running on this node
+			if err != nil {
+				return nil, err
+			}
 
-		_, okZone := node.GetLabels()[scheduler.ZoneLabel] //Node has no zone info (maybe zone is down or a control node) - CANNOT SCHEDULE VREPS on a pod running on this node
+			_, okZone := node.GetLabels()[scheduler.ZoneLabel] //Node has no zone info (maybe zone is down or a control node) - CANNOT SCHEDULE VREPS on a pod running on this node
 
-		taints := node.Spec.Taints
+			taints := node.Spec.Taints
 
-		if (!ok || !unschedulable) && !node.Spec.Unschedulable && okZone && !contains(taints, noexec) && !contains(taints, nosched) { //Pod has no annotation or not annotated as unschedulable and not on an unschedulable node, so add to feasible
-			schedulablePods = append(schedulablePods, podId)
+			if (!ok || !unschedulable) && !node.Spec.Unschedulable && okZone && !contains(taints, noexec) && !contains(taints, nosched) { //Pod has no annotation or not annotated as unschedulable and not on an unschedulable node, so add to feasible
+				schedulablePods = append(schedulablePods, podId)
+			}
 		}
 	}
 
@@ -282,11 +284,13 @@ func (s *stateBuilder) State(reserved map[types.NamespacedName]map[string]int32)
 				return err == nil, nil
 			})
 
-			nodeName := pod.Spec.NodeName       //node name for this pod
-			zoneName := nodeToZoneMap[nodeName] //zone name for this pod
-			podSpread[vpod.GetKey()][podName] = podSpread[vpod.GetKey()][podName] + vreplicas
-			nodeSpread[vpod.GetKey()][nodeName] = nodeSpread[vpod.GetKey()][nodeName] + vreplicas
-			zoneSpread[vpod.GetKey()][zoneName] = zoneSpread[vpod.GetKey()][zoneName] + vreplicas
+			if pod != nil {
+				nodeName := pod.Spec.NodeName       //node name for this pod
+				zoneName := nodeToZoneMap[nodeName] //zone name for this pod
+				podSpread[vpod.GetKey()][podName] = podSpread[vpod.GetKey()][podName] + vreplicas
+				nodeSpread[vpod.GetKey()][nodeName] = nodeSpread[vpod.GetKey()][nodeName] + vreplicas
+				zoneSpread[vpod.GetKey()][zoneName] = zoneSpread[vpod.GetKey()][zoneName] + vreplicas
+			}
 		}
 	}
 
@@ -305,11 +309,13 @@ func (s *stateBuilder) State(reserved map[types.NamespacedName]map[string]int32)
 					return err == nil, nil
 				})
 
-				nodeName := pod.Spec.NodeName       //node name for this pod
-				zoneName := nodeToZoneMap[nodeName] //zone name for this pod
-				podSpread[key][podName] = podSpread[key][podName] + rvreplicas
-				nodeSpread[key][nodeName] = nodeSpread[key][nodeName] + rvreplicas
-				zoneSpread[key][zoneName] = zoneSpread[key][zoneName] + rvreplicas
+				if pod != nil {
+					nodeName := pod.Spec.NodeName       //node name for this pod
+					zoneName := nodeToZoneMap[nodeName] //zone name for this pod
+					podSpread[key][podName] = podSpread[key][podName] + rvreplicas
+					nodeSpread[key][nodeName] = nodeSpread[key][nodeName] + rvreplicas
+					zoneSpread[key][zoneName] = zoneSpread[key][zoneName] + rvreplicas
+				}
 			}
 
 			free, last = s.updateFreeCapacity(free, last, podName, rvreplicas)

